dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/SystemActionApiFireCommandFactory.java

```diff
@@ -243,4 +281,347 @@ private Contentlet firePublishWithSave(final Contentlet contentlet,
+                        ", was checkin but does not have unpublish, so a unpublish will be fired (could autoassign if needed)");
+
+                contentletAPI.unpublish(checkinContentlet,
+                        user, dependencies.isRespectAnonymousPermissions());
+            }
+
+            return checkinContentlet;
+        }
+    }
+
+    //////////////////////////
+    /**
+     * Implements a {@link SystemActionApiFireCommand} that does an archive to cover the {@link com.dotmarketing.portlets.workflows.business.WorkflowAPI.SystemAction#ARCHIVE} system action
+     */
+    private class ArchiveSystemActionApiFireCommandImpl implements SystemActionApiFireCommand {
+
+        @WrapInTransaction
+        @Override
+        public Contentlet fire(final Contentlet contentlet, final boolean needSave, final ContentletDependencies dependencies)
+                throws DotDataException, DotSecurityException {
+
+            Logger.info(this, "The contentlet : " + contentlet.getTitle()
+                    + ", was fired by default action: " + dependencies.getWorkflowActionId() +
+                    ", however this action has not any archive content actionlet, so the archive api call is being triggered as part of the request");
+
+            final String actionId       = UtilMethods.isSet(dependencies.getWorkflowActionId())?
+                    dependencies.getWorkflowActionId():contentlet.getActionId();
+            final User   user           = dependencies.getModUser();
+
+            if(UtilMethods.isSet(actionId)) {
+                contentlet.setActionId(actionId);
+            }
+
+            if(UtilMethods.isSet(dependencies.getWorkflowActionComments())){
+                contentlet.setStringProperty(Contentlet.WORKFLOW_COMMENTS_KEY, dependencies.getWorkflowActionComments());
+            }
+
+            if(UtilMethods.isSet(dependencies.getWorkflowAssignKey())){
+                contentlet.setStringProperty(Contentlet.WORKFLOW_ASSIGN_KEY, dependencies.getWorkflowAssignKey());
+            }
+
+            if (needSave) {
+
+                return fireArchiveWithSave(contentlet, dependencies, actionId, user);
+            }
+
+            Logger.info(this, "The contentlet : " + contentlet.getTitle()
+                    + ", on the action id: " + actionId +
+                    ", will do an archive");
+
+            contentletAPI.archive(contentlet, user, dependencies.isRespectAnonymousPermissions());
+
+            return contentlet;
+        }
+    }
+
+    private Contentlet fireArchiveWithSave(final Contentlet contentlet,
+                                             final ContentletDependencies dependencies,
+                                             final String actionId, final User user) throws DotDataException, DotSecurityException {
+
+        final String disableWorkflow         = Contentlet.DISABLE_WORKFLOW;
+        final boolean hasArchiveActionlet  = UtilMethods.isSet(actionId)?
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Ternary operators that can be simplified with || or &&](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4324168)